In [1]:
import numpy as np 
import pandas as pd 

In [2]:
import json 
from sklearn import preprocessing

In [3]:
import matplotlib.pyplot as plt

In [131]:
import math as m 

In [131] used -25.0 MiB RAM in 0.11s (system mean cpu 20%, single max cpu 34%), peaked 0.0 MiB above final usage, current RAM usage now 17.8 MiB


In [4]:
%load_ext ipython_memory_usage

Enabling IPython Memory Usage, use %imu_start to begin, %imu_stop to end


In [5]:
%imu_start 

'IPython Memory Usage started'

In [5] used 0.4 MiB RAM in 0.12s (system mean cpu 0%, single max cpu 0%), peaked 0.0 MiB above final usage, current RAM usage now 153.6 MiB


### EDA de la dataFrame 'Athlete_events'

In [6]:
df_athlete=pd.read_csv('data/athlete_events.csv',sep=',')

df_athlete

Name Sex   Age  Height  Weight  \
0                      A Dijiang   M  24.0  180.00    80.0   
1                       A Lamusi   M  23.0  170.00    60.0   
2            Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3           Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4       Christine Jacoba Aaftink   F  21.0  185.00    82.0   
...                          ...  ..   ...     ...     ...   
282985              ZOU Jingyuan   M  23.0    1.58     NaN   
282986          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282987            ZUEV Alexander   M  25.0    1.93     NaN   
282988          ZVEREV Alexander   M  24.0    1.98     NaN   
282989              ZYKOVA Yulia   F  26.0     NaN     NaN   

                              Team  NOC        Games  Year  Season       City  \
0                            China  CHN  1992 Summer  1992  Summer  Barcelona   
1                            China  CHN  2012 Summer  2012  Summer     London   
2                          Denmark  DEN  1920 Summer  1920  Summer  Antwerpen   
3                   Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris   
4                      Netherlands  NED  1988 Winter  1988  Winter    Calgary   
...                            ...  ...          ...   ...     ...        ...   
282985  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282986                       Spain  ESP   Tokyo 2020  2020  Summer      Tokyo   
282987                         ROC  ROC   Tokyo 2020  2020  Summer      Tokyo   
282988                     Germany  GER   Tokyo 2020  2020  Summer      Tokyo   
282989                         ROC  ROC   Tokyo 2020  2020  Summer      Tokyo   

                      Sport                             Event   Medal  
0                Basketball       Basketball Men's Basketball     NaN  
1                      Judo      Judo Men's Extra-Lightweight     NaN  
2                  Football           Football Men's Football     NaN  
3                Tug-Of-War       Tug-Of-War Men's Tug-Of-War    Gold  
4             Speed Skating  Speed Skating Women's 500 metres     NaN  
...                     ...                               ...     ...  
282985  Artistic Gymnastics               Men's Parallel Bars    Gold  
282986             Football                               Men  Silver  
282987       3x3 Basketball                               Men  Silver  
282988               Tennis                     Men's Singles    Gold  
282989             Shooting       50m Rifle 3 Positions Women  Silver  

[282990 rows x 14 columns]

In [6] used 114.9 MiB RAM in 5.27s (system mean cpu 15%, single max cpu 90%), peaked 0.0 MiB above final usage, current RAM usage now 268.5 MiB


### DataFrame dictionnary

In [125]:
df_dictionnary=pd.read_csv('data/dictionary.csv')

df_dictionnary=df_dictionnary.rename(columns={'GDP per Capita':'PIB/habitant','Code':'NOC'})

df_dictionnary.isna().sum()

Country          0
NOC              0
Population       5
PIB/habitant    25
dtype: int64

In [125] used -20.0 MiB RAM in 1.26s (system mean cpu 32%, single max cpu 60%), peaked 0.0 MiB above final usage, current RAM usage now 27.3 MiB


### On cherche les athlètes qui n'ont pas de team et ceux des îles TUVALU

In [8]:
df_athlete[df_athlete['NOC']=='UNK']


Name Sex  Age  Height  Weight     Team  NOC        Games  \
61080   Fritz Eccard   M  NaN     NaN     NaN  Unknown  UNK  1912 Summer   
130721     A. Laffen   M  NaN     NaN     NaN  Unknown  UNK  1912 Summer   

        Year  Season       City             Sport  \
61080   1912  Summer  Stockholm  Art Competitions   
130721  1912  Summer  Stockholm  Art Competitions   

                                      Event Medal  
61080   Art Competitions Mixed Architecture   NaN  
130721  Art Competitions Mixed Architecture   NaN

In [8] used 3.3 MiB RAM in 0.15s (system mean cpu 13%, single max cpu 70%), peaked 0.0 MiB above final usage, current RAM usage now 201.2 MiB


Fritez Eccard a participé dans l'équipe Suédoise et A.Laffen .... aucunes traces je sais pas qui est ce mec 

In [9]:
df_athlete[df_athlete['NOC']=='ROT'].nunique()


Name      10
Sex        2
Age        7
Height     7
Weight     9
Team       1
NOC        1
Games      1
Year       1
Season     1
City       1
Sport      3
Event     12
Medal      0
dtype: int64

In [9] used 1.0 MiB RAM in 0.14s (system mean cpu 22%, single max cpu 57%), peaked 0.0 MiB above final usage, current RAM usage now 202.2 MiB


12 participations parmi 10 athlètes qui ont participé aux JO ces dernières années sous la team des athlètes réfugiés. 

In [10]:
df_athlete=df_athlete.query(" NOC !='ROC' and NOC != 'UNK' ")

In [10] used 26.5 MiB RAM in 0.18s (system mean cpu 11%, single max cpu 60%), peaked 0.0 MiB above final usage, current RAM usage now 228.7 MiB


## Base données région 

In [63]:
df_regions=pd.read_csv('data/noc_regions.csv')

df_regions=df_regions.rename(columns={'region':'Country'})
df_regions

NOC      Country                 notes
0    AFG  Afghanistan                   NaN
1    AHO      Curacao  Netherlands Antilles
2    ALB      Albania                   NaN
3    ALG      Algeria                   NaN
4    AND      Andorra                   NaN
..   ...          ...                   ...
225  YEM        Yemen                   NaN
226  YMD        Yemen           South Yemen
227  YUG       Serbia            Yugoslavia
228  ZAM       Zambia                   NaN
229  ZIM     Zimbabwe                   NaN

[230 rows x 3 columns]

In [63] used -1.2 MiB RAM in 0.13s (system mean cpu 16%, single max cpu 24%), peaked 0.0 MiB above final usage, current RAM usage now 27.0 MiB


In [64]:
df_regions.nunique()

NOC        230
Country    206
notes       21
dtype: int64

In [64] used 1.5 MiB RAM in 0.12s (system mean cpu 17%, single max cpu 24%), peaked 0.0 MiB above final usage, current RAM usage now 28.5 MiB


Il y a des régions qui ont le même NOC 

In [65]:
duplicates=[df_regions.duplicated(subset='Country')]

In [65] used 0.3 MiB RAM in 0.11s (system mean cpu 13%, single max cpu 22%), peaked 0.0 MiB above final usage, current RAM usage now 28.8 MiB


In [66]:
df_regions[df_regions.duplicated(subset='Country')]

NOC         Country                   notes
12   AUS       Australia                     NaN
54   CZE  Czech Republic                     NaN
76   GDR         Germany                     NaN
79   GER         Germany                     NaN
81   GRE          Greece                     NaN
88   HKG           China               Hong Kong
126  MAS        Malaysia                     NaN
143  NBO        Malaysia            North Borneo
147  NFL          Canada            Newfoundland
171  RUS          Russia                     NaN
173  SAA         Germany                     NaN
185  SRB          Serbia                     NaN
197  TCH  Czech Republic                     NaN
208  TUV             NaN                  Tuvalu
210  UAR           Syria    United Arab Republic
213  UNK             NaN                 Unknown
214  URS          Russia                     NaN
222  VNM         Vietnam                     NaN
223  WIF        Trinidad  West Indies Federation
225  YEM           Yemen                     NaN
226  YMD           Yemen             South Yemen
227  YUG          Serbia              Yugoslavia
229  ZIM        Zimbabwe                     NaN

In [66] used 0.4 MiB RAM in 0.12s (system mean cpu 13%, single max cpu 23%), peaked 0.0 MiB above final usage, current RAM usage now 29.2 MiB


In [67]:
df_regions.isnull().sum()

NOC          0
Country      3
notes      209
dtype: int64

In [67] used 0.2 MiB RAM in 0.11s (system mean cpu 11%, single max cpu 21%), peaked 0.0 MiB above final usage, current RAM usage now 29.5 MiB


In [68]:
df_regions[df_regions['Country'].isnull()]

NOC Country                 notes
168  ROT     NaN  Refugee Olympic Team
208  TUV     NaN                Tuvalu
213  UNK     NaN               Unknown

In [68] used 0.0 MiB RAM in 0.12s (system mean cpu 16%, single max cpu 24%), peaked 0.0 MiB above final usage, current RAM usage now 29.5 MiB


Il y a  3 des regions qui ne sont pas attribuées, ROT ( refugee olympic team)  , TUVALU ( 9 archipels qui constituent un État ) et UNK qui correspond à Unknown 

### On ajoute les iles TUVALU 

In [69]:
df_regions[df_regions['Country'].isnull()]

NOC Country                 notes
168  ROT     NaN  Refugee Olympic Team
208  TUV     NaN                Tuvalu
213  UNK     NaN               Unknown

In [69] used -2.0 MiB RAM in 0.11s (system mean cpu 16%, single max cpu 25%), peaked 0.0 MiB above final usage, current RAM usage now 27.5 MiB


In [70]:
df_regions=df_regions.query(" NOC != 'ROT' and NOC != 'UNK' ")

df_regions.isnull().sum()

NOC          0
Country      1
notes      209
dtype: int64

In [70] used 2.4 MiB RAM in 0.14s (system mean cpu 12%, single max cpu 20%), peaked 0.0 MiB above final usage, current RAM usage now 29.9 MiB


In [71]:
df_regions['Country']=df_regions['Country'].fillna('Tuvalu')

In [71] used -5.0 MiB RAM in 0.12s (system mean cpu 14%, single max cpu 23%), peaked 0.0 MiB above final usage, current RAM usage now 24.9 MiB


<ipython-input-71-0f7f308fe64b>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_regions['Country']=df_regions['Country'].fillna('Tuvalu')


In [73]:
df_regions[df_regions['Country']=='Tuvalu']

NOC Country   notes
208  TUV  Tuvalu  Tuvalu

In [73] used 2.1 MiB RAM in 0.13s (system mean cpu 16%, single max cpu 25%), peaked 0.0 MiB above final usage, current RAM usage now 43.1 MiB


In [74]:
df_regions.isnull().sum()

NOC          0
Country      0
notes      209
dtype: int64

In [74] used 0.4 MiB RAM in 0.11s (system mean cpu 17%, single max cpu 26%), peaked 0.0 MiB above final usage, current RAM usage now 43.6 MiB


# On va merge les dataframes 

## Merge en left 

In [75]:
df_complete=pd.merge(pd.merge(df_athlete,df_regions,on='NOC',how='left'),df_dictionnary,on='NOC',how='left')
#df_complete=df_complete.drop(labels='',axis=1)

df_complete

Name Sex   Age  Height  Weight  \
0                      A Dijiang   M  24.0  180.00    80.0   
1                       A Lamusi   M  23.0  170.00    60.0   
2            Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3           Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4       Christine Jacoba Aaftink   F  21.0  185.00    82.0   
...                          ...  ..   ...     ...     ...   
282622        ZOLOTIC Anastasija   F  19.0    1.80     NaN   
282623              ZOU Jingyuan   M  23.0    1.58     NaN   
282624              ZOU Jingyuan   M  23.0    1.58     NaN   
282625          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282626          ZVEREV Alexander   M  24.0    1.98     NaN   

                              Team  NOC        Games  Year  Season       City  \
0                            China  CHN  1992 Summer  1992  Summer  Barcelona   
1                            China  CHN  2012 Summer  2012  Summer     London   
2                          Denmark  DEN  1920 Summer  1920  Summer  Antwerpen   
3                   Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris   
4                      Netherlands  NED  1988 Winter  1988  Winter    Calgary   
...                            ...  ...          ...   ...     ...        ...   
282622    United States of America  USA   Tokyo 2020  2020  Summer      Tokyo   
282623  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282624  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282625                       Spain  ESP   Tokyo 2020  2020  Summer      Tokyo   
282626                     Germany  GER   Tokyo 2020  2020  Summer      Tokyo   

                      Sport                             Event   Medal  \
0                Basketball       Basketball Men's Basketball     NaN   
1                      Judo      Judo Men's Extra-Lightweight     NaN   
2                  Football           Football Men's Football     NaN   
3                Tug-Of-War       Tug-Of-War Men's Tug-Of-War    Gold   
4             Speed Skating  Speed Skating Women's 500 metres     NaN   
...                     ...                               ...     ...   
282622            Taekwondo                       Women -57kg    Gold   
282623  Artistic Gymnastics                        Men's Team  Bronze   
282624  Artistic Gymnastics               Men's Parallel Bars    Gold   
282625             Football                               Men  Silver   
282626               Tennis                     Men's Singles    Gold   

          Country_x notes      Country_y    Population  PIB/habitant  
0             China   NaN          China  1.371220e+09   8027.683810  
1             China   NaN          China  1.371220e+09   8027.683810  
2           Denmark   NaN        Denmark  5.676002e+06  51989.293471  
3           Denmark   NaN        Denmark  5.676002e+06  51989.293471  
4       Netherlands   NaN    Netherlands  1.693652e+07  44299.768085  
...             ...   ...            ...           ...           ...  
282622          USA   NaN  United States  3.214188e+08  56115.718426  
282623        China   NaN          China  1.371220e+09   8027.683810  
282624        China   NaN          China  1.371220e+09   8027.683810  
282625        Spain   NaN          Spain  4.641827e+07  25831.582305  
282626      Germany   NaN        Germany  8.141314e+07  41313.313995  

[282627 rows x 19 columns]

In [75] used 130.4 MiB RAM in 0.39s (system mean cpu 20%, single max cpu 50%), peaked 0.0 MiB above final usage, current RAM usage now 174.0 MiB


In [76]:
df_complete=df_complete.query(" NOC !='ROC' ")

In [76] used -46.5 MiB RAM in 0.23s (system mean cpu 13%, single max cpu 27%), peaked 0.0 MiB above final usage, current RAM usage now 127.5 MiB


In [77]:
#df_complete.info()

In [77] used 0.9 MiB RAM in 0.10s (system mean cpu 0%, single max cpu 0%), peaked 0.0 MiB above final usage, current RAM usage now 128.3 MiB


282 627 lignes , c'est normal on a enlevé les  réfugiés

In [79]:
df_complete.loc[(df_complete['Medal'].isna(),'Medal')]='Pas_medaille'



In [79] used -57.8 MiB RAM in 0.14s (system mean cpu 15%, single max cpu 24%), peaked 32.0 MiB above final usage, current RAM usage now 66.9 MiB


## Encoding test 

In [80]:
dictio_score_medaille={'Pas_medaille':0,
                      'Bronze':1,
                      'Silver':2,
                      'Gold':3}

df_complete['Score_medaille']=df_complete['Medal'].map(dictio_score_medaille)

df_complete

Name Sex   Age  Height  Weight  \
0                      A Dijiang   M  24.0  180.00    80.0   
1                       A Lamusi   M  23.0  170.00    60.0   
2            Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3           Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4       Christine Jacoba Aaftink   F  21.0  185.00    82.0   
...                          ...  ..   ...     ...     ...   
282622        ZOLOTIC Anastasija   F  19.0    1.80     NaN   
282623              ZOU Jingyuan   M  23.0    1.58     NaN   
282624              ZOU Jingyuan   M  23.0    1.58     NaN   
282625          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282626          ZVEREV Alexander   M  24.0    1.98     NaN   

                              Team  NOC        Games  Year  Season       City  \
0                            China  CHN  1992 Summer  1992  Summer  Barcelona   
1                            China  CHN  2012 Summer  2012  Summer     London   
2                          Denmark  DEN  1920 Summer  1920  Summer  Antwerpen   
3                   Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris   
4                      Netherlands  NED  1988 Winter  1988  Winter    Calgary   
...                            ...  ...          ...   ...     ...        ...   
282622    United States of America  USA   Tokyo 2020  2020  Summer      Tokyo   
282623  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282624  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282625                       Spain  ESP   Tokyo 2020  2020  Summer      Tokyo   
282626                     Germany  GER   Tokyo 2020  2020  Summer      Tokyo   

                      Sport                             Event         Medal  \
0                Basketball       Basketball Men's Basketball  Pas_medaille   
1                      Judo      Judo Men's Extra-Lightweight  Pas_medaille   
2                  Football           Football Men's Football  Pas_medaille   
3                Tug-Of-War       Tug-Of-War Men's Tug-Of-War          Gold   
4             Speed Skating  Speed Skating Women's 500 metres  Pas_medaille   
...                     ...                               ...           ...   
282622            Taekwondo                       Women -57kg          Gold   
282623  Artistic Gymnastics                        Men's Team        Bronze   
282624  Artistic Gymnastics               Men's Parallel Bars          Gold   
282625             Football                               Men        Silver   
282626               Tennis                     Men's Singles          Gold   

          Country_x notes      Country_y    Population  PIB/habitant  \
0             China   NaN          China  1.371220e+09   8027.683810   
1             China   NaN          China  1.371220e+09   8027.683810   
2           Denmark   NaN        Denmark  5.676002e+06  51989.293471   
3           Denmark   NaN        Denmark  5.676002e+06  51989.293471   
4       Netherlands   NaN    Netherlands  1.693652e+07  44299.768085   
...             ...   ...            ...           ...           ...   
282622          USA   NaN  United States  3.214188e+08  56115.718426   
282623        China   NaN          China  1.371220e+09   8027.683810   
282624        China   NaN          China  1.371220e+09   8027.683810   
282625        Spain   NaN          Spain  4.641827e+07  25831.582305   
282626      Germany   NaN        Germany  8.141314e+07  41313.313995   

        Score_medaille  
0                    0  
1                    0  
2                    0  
3                    3  
4                    0  
...                ...  
282622               3  
282623               1  
282624               3  
282625               2  
282626               3  

[282627 rows x 20 columns]

In [80] used -26.6 MiB RAM in 0.15s (system mean cpu 15%, single max cpu 24%), peaked 0.0 MiB above final usage, current RAM usage now 40.3 MiB


In [26]:
#Marche aussi 
''' 
medaille_categories=['Pas_medaille','Bronze','Silver','Gold']
encode_medaille=preprocessing.OrdinalEncoder(categories=[medaille_categories])

df_complete['enc_medal']=encode_medaille.fit_transform(df_complete[['Medal']])

df_complete
''' 

" \nmedaille_categories=['Pas_medaille','Bronze','Silver','Gold']\nencode_medaille=preprocessing.OrdinalEncoder(categories=[medaille_categories])\n\ndf_complete['enc_medal']=encode_medaille.fit_transform(df_complete[['Medal']])\n\ndf_complete\n"

In [26] used -3.6 MiB RAM in 0.11s (system mean cpu 17%, single max cpu 100%), peaked 3.6 MiB above final usage, current RAM usage now 333.0 MiB


## Analyse valeurs nulles 

In [81]:
df_complete.isna().sum()

Name                   0
Sex                  156
Age                 9628
Height             67142
Weight             74386
Team                   0
NOC                    0
Games                  0
Year                   0
Season                 0
City                   0
Sport                156
Event               9278
Medal                  0
Country_x            419
notes             277510
Country_y          26221
Population         27593
PIB/habitant       35919
Score_medaille         0
dtype: int64

In [81] used 51.4 MiB RAM in 0.26s (system mean cpu 22%, single max cpu 46%), peaked 0.0 MiB above final usage, current RAM usage now 91.7 MiB


156 sports où on sait pas ce que c'est .... et des country (26 221) qui ne sont pas attribués non plus 

Pour les Country , on va merge 

In [114]:
#df_complete_test=df_complete.copy()

df_complete=pd.merge(df_complete,df_regions,on='NOC',how='left')

df_complete['Country']=df_complete['Country_y'].fillna(df_complete['Country_x'])

In [114] used 48.4 MiB RAM in 0.31s (system mean cpu 21%, single max cpu 60%), peaked 0.0 MiB above final usage, current RAM usage now 143.3 MiB


In [115]:
df_complete= df_complete.drop(['Country_x', 'Country_y'], axis=1)

In [115] used 37.4 MiB RAM in 0.21s (system mean cpu 20%, single max cpu 29%), peaked 0.0 MiB above final usage, current RAM usage now 180.6 MiB


In [116]:
liste_pays=df_regions['Country'].tolist()
liste_NOC=df_regions['NOC'].tolist()

dictio_NOC_pays=dict(zip(liste_NOC,liste_pays))

dictio_NOC_pays['SGP']='Singapour'
dictio_NOC_pays['LBN']='Liban'


df_complete['Country']=df_complete['Country'].fillna(df_complete['NOC'].map(dictio_NOC_pays))
#liste_region

#df_complete_test.isna().sum()

Name                   0
Sex                  156
Age                 9628
Height             67142
Weight             74386
Team                   0
NOC                    0
Games                  0
Year                   0
Season                 0
City                   0
Sport                156
Event               9278
Medal                  0
notes_x           277510
Population         27593
PIB/habitant       35919
Score_medaille         0
Country               41
notes_y           277510
dtype: int64

In [116] used -85.7 MiB RAM in 0.32s (system mean cpu 17%, single max cpu 36%), peaked 0.0 MiB above final usage, current RAM usage now 94.9 MiB


In [109]:
test=df_complete_test[df_complete_test['Country'].isna()]

#test['Country']=test['Country'].fillna(test['NOC'].map(dictio_NOC_pays))

test.groupby('NOC')['Name'].count().reset_index()

NOC  Name
0  EOR    29
1  LBN     6
2  ROT    12

In [109] used 3.9 MiB RAM in 0.14s (system mean cpu 17%, single max cpu 26%), peaked 0.0 MiB above final usage, current RAM usage now 99.8 MiB


Il y a quatre NOC qui ne correspondent à aucun pays dans notre EOR : réfugiés , ROT réfugiés aussi , LBN c'est le Liban (Lebanon en anglais )

In [119]:
df_complete=df_complete.query(" NOC !='ROT' and NOC !='EOR' ")



Name                   0
Sex                  156
Age                 9628
Height             67116
Weight             74357
Team                   0
NOC                    0
Games                  0
Year                   0
Season                 0
City                   0
Sport                156
Event               9249
Medal                  0
notes_x           277469
Population         27552
PIB/habitant       35878
Score_medaille         0
Country                0
notes_y           277469
dtype: int64

In [119] used 4.6 MiB RAM in 0.34s (system mean cpu 22%, single max cpu 37%), peaked 0.0 MiB above final usage, current RAM usage now 138.5 MiB


In [121]:
df_complete=df_complete.drop(['notes_x','notes_y'],axis=1)

In [121] used 90.6 MiB RAM in 0.22s (system mean cpu 18%, single max cpu 28%), peaked 0.0 MiB above final usage, current RAM usage now 128.5 MiB


## Opérations sur les sports 

In [30]:
liste_sports=df_complete['Sport'].unique().tolist()

print(liste_sports)

['Basketball', 'Judo', 'Football', 'Tug-Of-War', 'Speed Skating', 'Cross Country Skiing', 'Athletics', 'Ice Hockey', 'Swimming', 'Badminton', 'Sailing', 'Biathlon', 'Gymnastics', 'Art Competitions', 'Alpine Skiing', 'Handball', 'Weightlifting', 'Wrestling', 'Luge', 'Water Polo', 'Hockey', 'Rowing', 'Bobsleigh', 'Fencing', 'Equestrianism', 'Shooting', 'Boxing', 'Taekwondo', 'Cycling', 'Diving', 'Canoeing', 'Tennis', 'Modern Pentathlon', 'Figure Skating', 'Golf', 'Softball', 'Archery', 'Volleyball', 'Synchronized Swimming', 'Table Tennis', 'Nordic Combined', 'Baseball', 'Rhythmic Gymnastics', 'Freestyle Skiing', 'Rugby Sevens', 'Trampolining', 'Beach Volleyball', 'Triathlon', 'Ski Jumping', 'Curling', 'Snowboarding', 'Rugby', 'Short Track Speed Skating', 'Skeleton', 'Lacrosse', 'Polo', 'Cricket', 'Racquets', 'Motorboating', 'Military Ski Patrol', 'Croquet', 'Jeu De Paume', 'Roque', 'Alpinism', 'Basque Pelota', 'Aeronautics', 'Cycling Road', 'Artistic Gymnastics', 'Karate', 'Trampoline Gy

In [31]:
table_sport=df_complete.groupby('Sport')['Name'].count().reset_index()

table_sport

Sport  Name
0   3x3 Basketball    56
1      Aeronautics     1
2    Alpine Skiing  8829
3         Alpinism    25
4          Archery  2461
..             ...   ...
79      Tug-Of-War   170
80      Volleyball  3666
81      Water Polo  4119
82   Weightlifting  4129
83       Wrestling  7426

[84 rows x 2 columns]

In [31] used 2.4 MiB RAM in 0.16s (system mean cpu 28%, single max cpu 83%), peaked 0.0 MiB above final usage, current RAM usage now 337.4 MiB


In [122]:
df_complete=df_complete[df_complete['Year']>=2000]

In [122] used -39.2 MiB RAM in 0.39s (system mean cpu 20%, single max cpu 36%), peaked 0.0 MiB above final usage, current RAM usage now 89.2 MiB


In [123]:
df_complete

Name Sex   Age  Height  Weight  \
1                       A Lamusi   M  23.0  170.00    60.0   
28              Antti Sami Aalto   M  26.0  186.00    96.0   
31             Jyri Tapani Aalto   M  31.0  172.00    70.0   
33            Minna Maarit Aalto   F  34.0  159.00    55.5   
40      Juhamatti Tapio Aaltonen   M  28.0  184.00    85.0   
...                          ...  ..   ...     ...     ...   
282622        ZOLOTIC Anastasija   F  19.0    1.80     NaN   
282623              ZOU Jingyuan   M  23.0    1.58     NaN   
282624              ZOU Jingyuan   M  23.0    1.58     NaN   
282625          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282626          ZVEREV Alexander   M  24.0    1.98     NaN   

                              Team  NOC        Games  Year  Season  \
1                            China  CHN  2012 Summer  2012  Summer   
28                         Finland  FIN  2002 Winter  2002  Winter   
31                         Finland  FIN  2000 Summer  2000  Summer   
33                         Finland  FIN  2000 Summer  2000  Summer   
40                         Finland  FIN  2014 Winter  2014  Winter   
...                            ...  ...          ...   ...     ...   
282622    United States of America  USA   Tokyo 2020  2020  Summer   
282623  People's Republic of China  CHN   Tokyo 2020  2020  Summer   
282624  People's Republic of China  CHN   Tokyo 2020  2020  Summer   
282625                       Spain  ESP   Tokyo 2020  2020  Summer   
282626                     Germany  GER   Tokyo 2020  2020  Summer   

                  City                Sport                         Event  \
1               London                 Judo  Judo Men's Extra-Lightweight   
28      Salt Lake City           Ice Hockey   Ice Hockey Men's Ice Hockey   
31              Sydney            Badminton       Badminton Men's Singles   
33              Sydney              Sailing    Sailing Women's Windsurfer   
40               Sochi           Ice Hockey   Ice Hockey Men's Ice Hockey   
...                ...                  ...                           ...   
282622           Tokyo            Taekwondo                   Women -57kg   
282623           Tokyo  Artistic Gymnastics                    Men's Team   
282624           Tokyo  Artistic Gymnastics           Men's Parallel Bars   
282625           Tokyo             Football                           Men   
282626           Tokyo               Tennis                 Men's Singles   

               Medal    Population  PIB/habitant  Score_medaille  \
1       Pas_medaille  1.371220e+09   8027.683810               0   
28      Pas_medaille  5.482013e+06  42311.036231               0   
31      Pas_medaille  5.482013e+06  42311.036231               0   
33      Pas_medaille  5.482013e+06  42311.036231               0   
40            Bronze  5.482013e+06  42311.036231               1   
...              ...           ...           ...             ...   
282622          Gold  3.214188e+08  56115.718426               3   
282623        Bronze  1.371220e+09   8027.683810               1   
282624          Gold  1.371220e+09   8027.683810               3   
282625        Silver  4.641827e+07  25831.582305               2   
282626          Gold  8.141314e+07  41313.313995               3   

              Country  
1               China  
28            Finland  
31            Finland  
33            Finland  
40            Finland  
...               ...  
282622  United States  
282623          China  
282624          China  
282625          Spain  
282626        Germany  

[96730 rows x 18 columns]

In [123] used 5.8 MiB RAM in 0.15s (system mean cpu 19%, single max cpu 29%), peaked 0.0 MiB above final usage, current RAM usage now 95.0 MiB


## Opérations sur les Ages

In [135]:
table_sport_age=df_complete.groupby('Sport')['Age'].mean().apply(m.ceil).reset_index()

table_sport_age

Sport  Age
0        3x3 Basketball   30
1         Alpine Skiing   25
2               Archery   27
3   Artistic Gymnastics   25
4     Artistic Swimming   24
..                  ...  ...
64            Triathlon   29
65           Volleyball   27
66           Water Polo   27
67        Weightlifting   25
68            Wrestling   27

[69 rows x 2 columns]

In [135] used 1.2 MiB RAM in 0.14s (system mean cpu 18%, single max cpu 30%), peaked 0.0 MiB above final usage, current RAM usage now 47.5 MiB


In [137]:
liste_sport=table_sport_age['Sport'].tolist()
liste_age=table_sport_age['Age'].tolist()

dictio_age_sport=dict(zip(liste_sport,liste_age))

In [137] used -22.9 MiB RAM in 0.11s (system mean cpu 13%, single max cpu 22%), peaked 0.0 MiB above final usage, current RAM usage now 24.7 MiB


In [138]:
df_complete['Age']=df_complete['Age'].fillna(df_complete['Sport'].map(dictio_age_sport))

df_complete.isna().sum()

Name                  0
Sex                 156
Age                 156
Height             7658
Weight            12694
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport               156
Event              9249
Medal                 0
Population         2855
PIB/habitant       5741
Score_medaille        0
Country               0
dtype: int64

In [138] used 31.7 MiB RAM in 0.23s (system mean cpu 14%, single max cpu 33%), peaked 0.0 MiB above final usage, current RAM usage now 56.4 MiB


In [142]:
df_complete[df_complete['Age'].isna()].groupby('Medal')['Name'].count().reset_index()

Medal  Name
0  Pas_medaille   156

In [142] used 1.0 MiB RAM in 0.12s (system mean cpu 13%, single max cpu 21%), peaked 0.0 MiB above final usage, current RAM usage now 30.5 MiB


In [140]:
#df_tokyo=pd.read_csv('data/athlete_events_tokyo_cleaned.csv')

#df_tokyo.isna().sum()

Name         0
Sex        159
Age        159
Height    7189
Team         0
NOC          0
Games        0
Year         0
Season       0
City         0
Sport      159
Medal     9492
Event     9492
dtype: int64

In [140] used 9.7 MiB RAM in 0.20s (system mean cpu 16%, single max cpu 25%), peaked 0.0 MiB above final usage, current RAM usage now 38.6 MiB


Enfaite il y a 156 personnes qui manquent prcq c'est les qu'on a pas les sport 

In [143]:
df_complete=df_complete.dropna(subset='Age')

In [143] used 34.6 MiB RAM in 0.22s (system mean cpu 14%, single max cpu 24%), peaked 0.0 MiB above final usage, current RAM usage now 65.0 MiB


In [144]:
df_complete.isna().sum()

Name                  0
Sex                   0
Age                   0
Height             7502
Weight            12538
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport                 0
Event              9093
Medal                 0
Population         2846
PIB/habitant       5731
Score_medaille        0
Country               0
dtype: int64

In [144] used 4.3 MiB RAM in 0.15s (system mean cpu 14%, single max cpu 24%), peaked 0.0 MiB above final usage, current RAM usage now 69.3 MiB


## Opérations sur les poids et les tailles 

In [155]:
table_taille=df_complete[df_complete['Height'].isna()==False].groupby(['Sport','Sex'])['Height'].mean().apply(m.ceil).reset_index()

liste_sport=table_taille['Sport'].tolist()
liste_sex=table_taille['Sex'].tolist()
liste_taille=table_taille['Height'].tolist()

liste_sport_sex=[]
for i in range(len(liste_sport)):
    liste_sport_sex.append((liste_sport[i],liste_sex[i]))
#print(liste_sport_sex)

dictio_sport_sex=dict(zip(liste_sport_sex,liste_taille))



{('3x3 Basketball', 'F'): 2,
 ('3x3 Basketball', 'M'): 2,
 ('Alpine Skiing', 'F'): 169,
 ('Alpine Skiing', 'M'): 181,
 ('Archery', 'F'): 168,
 ('Archery', 'M'): 180,
 ('Artistic Gymnastics', 'F'): 2,
 ('Artistic Gymnastics', 'M'): 2,
 ('Artistic Swimming', 'F'): 2,
 ('Athletics', 'F'): 169,
 ('Athletics', 'M'): 181,
 ('Badminton', 'F'): 168,
 ('Badminton', 'M'): 177,
 ('Baseball', 'M'): 184,
 ('Baseball/Softball', 'F'): 2,
 ('Baseball/Softball', 'M'): 2,
 ('Basketball', 'F'): 153,
 ('Basketball', 'M'): 168,
 ('Beach Volleyball', 'F'): 152,
 ('Beach Volleyball', 'M'): 164,
 ('Biathlon', 'F'): 167,
 ('Biathlon', 'M'): 180,
 ('Bobsleigh', 'F'): 174,
 ('Bobsleigh', 'M'): 185,
 ('Boxing', 'F'): 116,
 ('Boxing', 'M'): 170,
 ('Canoe Sprint', 'F'): 2,
 ('Canoe Sprint', 'M'): 2,
 ('Canoeing', 'F'): 171,
 ('Canoeing', 'M'): 183,
 ('Cross Country Skiing', 'F'): 167,
 ('Cross Country Skiing', 'M'): 180,
 ('Curling', 'F'): 168,
 ('Curling', 'M'): 181,
 ('Cycling', 'F'): 168,
 ('Cycling', 'M'): 180,

In [155] used 14.8 MiB RAM in 0.19s (system mean cpu 22%, single max cpu 38%), peaked 0.0 MiB above final usage, current RAM usage now 102.8 MiB


In [150]:
df_complete[df_complete['Sport']=='Wrestling']

Name Sex   Age  Height  Weight  \
81                               Fritz Aanes   M  22.0   187.0    89.0   
82                               Fritz Aanes   M  26.0   187.0    89.0   
186                     Stephen Anthony Abas   M  26.0   165.0    55.0   
288     Mohamed Ibrahim Abd El-Fatah Mohamed   M  22.0   185.0    98.0   
289     Mohamed Ibrahim Abd El-Fatah Mohamed   M  26.0   185.0    98.0   
...                                      ...  ..   ...     ...     ...   
282571                 YAZDANICHARATI Hassan   M  27.0     NaN     NaN   
282585                  YUSEIN Taybe Mustafa   F  30.0     NaN     NaN   
282597                     ZARE Amir Hossein   M  20.0     NaN     NaN   
282617                             ZHOU Qian   F  32.0     NaN     NaN   
282620                  ZHUMANAZAROVA Meerim   F  22.0     NaN     NaN   

                              Team  NOC        Games  Year  Season    City  \
81                          Norway  NOR  2000 Summer  2000  Summer  Sydney   
82                          Norway  NOR  2004 Summer  2004  Summer  Athina   
186                  United States  USA  2004 Summer  2004  Summer  Athina   
288                          Egypt  EGY  2000 Summer  2000  Summer  Sydney   
289                          Egypt  EGY  2004 Summer  2004  Summer  Athina   
...                            ...  ...          ...   ...     ...     ...   
282571    Islamic Republic of Iran  IRI   Tokyo 2020  2020  Summer   Tokyo   
282585                    Bulgaria  BUL   Tokyo 2020  2020  Summer   Tokyo   
282597    Islamic Republic of Iran  IRI   Tokyo 2020  2020  Summer   Tokyo   
282617  People's Republic of China  CHN   Tokyo 2020  2020  Summer   Tokyo   
282620                  Kyrgyzstan  KGZ   Tokyo 2020  2020  Summer   Tokyo   

            Sport                                           Event  \
81      Wrestling  Wrestling Men's Light-Heavyweight, Greco-Roman   
82      Wrestling  Wrestling Men's Light-Heavyweight, Greco-Roman   
186     Wrestling        Wrestling Men's Featherweight, Freestyle   
288     Wrestling  Wrestling Men's Light-Heavyweight, Greco-Roman   
289     Wrestling  Wrestling Men's Light-Heavyweight, Greco-Roman   
...           ...                                             ...   
282571  Wrestling                            Men's Freestyle 86kg   
282585  Wrestling                          Women's Freestyle 62kg   
282597  Wrestling                           Men's Freestyle 125kg   
282617  Wrestling                          Women's Freestyle 76kg   
282620  Wrestling                          Women's Freestyle 68kg   

               Medal    Population  PIB/habitant  Score_medaille  \
81      Pas_medaille  5.195921e+06  74400.369777               0   
82      Pas_medaille  5.195921e+06  74400.369777               0   
186           Silver  3.214188e+08  56115.718426               2   
288     Pas_medaille  9.150808e+07   3614.746766               0   
289     Pas_medaille  9.150808e+07   3614.746766               0   
...              ...           ...           ...             ...   
282571        Silver  7.910927e+07           NaN               2   
282585        Bronze  7.177991e+06   6993.477360               1   
282597        Bronze  7.910927e+07           NaN               1   
282617        Bronze  1.371220e+09   8027.683810               1   
282620        Bronze  5.957000e+06   1103.215352               1   

              Country  
81             Norway  
82             Norway  
186     United States  
288             Egypt  
289             Egypt  
...               ...  
282571           Iran  
282585       Bulgaria  
282597           Iran  
282617          China  
282620     Kyrgyzstan  

[1955 rows x 18 columns]

In [150] used 16.8 MiB RAM in 0.15s (system mean cpu 18%, single max cpu 31%), peaked 0.0 MiB above final usage, current RAM usage now 46.6 MiB
